### PR TITLE
Block Library: Add the Comments Pagination Next block

### LINKED ISSUE #35008: Post Comments Pagination Next block
## Description

Displays the next comments page link. This should be similar to the [Query Pagination Next block](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/query-pagination-next/).

This block is part of the Post Comments Loop block, for more context please visit this [Tracking Issue](https://github.com/WordPress/gutenberg/issues/34994) or its [GitHub project](https://github.com/WordPress/gutenberg/projects/61).

## Potential Settings

- **Typography:** This should be similar to the [Query Pagination Next block](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/query-pagination-next/).
- **Color settings:** This should be similar to the [Query Pagination Next block](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/query-pagination-next/)

## Requirements

- This block should only be available inside the [Post Comments Pagination Block](https://github.com/WordPress/gutenberg/issues/35007).
- If there are no next pages, we shouldn't show the button.

## References

This should be similar to the [Query Pagination Next block](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/query-pagination-next/).

### PR DESCRIPTION
**Pull request by cbravobernal**
## Description

We are adding the Next Comments block for FSE with this PR. Will close #35008 once is done.

## How has this been tested?

- [x] Create a Comment Query Loop inside a post or a page.
- [x] Add pagination.
- [x] Check arrows are working, including RTL version.
- [x] Check that if you are on the last page, the link is not appearing.

## Screenshots
https://loom.com/share/553a6762010747c7ba591a208c4a8a34

## Types of changes

- New feature. New core block.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/comments-pagination-next/edit.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+					tagName="span"
+					aria-label={ __( 'Next comments page link' ) }
+					placeholder={ __( 'Next Comments' ) }
+					value={ label }
+					onChange={ ( newLabel ) =>
+						setAttributes( { label: newLabel } )
+					}
+				/>
+				{ displayArrow && (
+					<span className={ `wp-block-query-pagination-next-arrow` }>
```
Comment (gziolo) [inline packages/block-library/src/comments-pagination-next/edit.js]: It looks like a copy and paste issue:

```suggestion
					<span className="wp-block-comments-pagination-next-arrow">
```

File: packages/block-library/src/comments-pagination-next/index.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+/**
+ * Renders the `core/comments-pagination-next` block on the server.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the next posts link for the query pagination.
```
Comment (gziolo) [inline packages/block-library/src/comments-pagination-next/index.php]: A copy and paste nitpicking:
```suggestion
 * @return string Returns the next comments link for the query pagination.
```

File: packages/block-library/src/query-pagination-next/edit.js [hunk 3]
```diff
@@ -17,28 +17,29 @@ export default function QueryPaginationNextEdit( {
...
-			/>
-			{ displayArrow && (
-				<span
-					className={ `wp-block-query-pagination-next-arrow is-arrow-${ paginationArrow }` }
-				>
-					{ displayArrow }
-				</span>
-			) }
-		</a>
+		<div { ...useBlockProps() }>
```
Comment (gziolo) [inline packages/block-library/src/query-pagination-next/edit.js]: This change looks unrelated. What’s the issue here? We should rather process it independently.

**Review by gziolo — COMMENTED** [review]
This PR depends on the PR that implements the Comments Pagination block so let’s wait with testing once everything is integrated. It looks great so far 👍

File: packages/block-library/src/query-pagination-next/edit.js (hunk 3 above)
Comment (cbravobernal) [reply packages/block-library/src/query-pagination-next/edit.js]: If we don't put a parent div there, the option of changing link color on the editor won't work. It is not working now on Query Pagination Next and Previous links. 😔

With a parent `div`, a link works.
Screenshot 2021-11-29 at 11 23 19

Without, it won't:
Screenshot 2021-11-29 at 11 24 12

Comment (gziolo) [reply packages/block-library/src/query-pagination-next/edit.js]: @ntsekouras, is it something on the radar? It looks like a bug fix worth considering for WP 5.9.

Comment (ntsekouras) [reply packages/block-library/src/query-pagination-next/edit.js]: Good catch! I opened a PR proposing to remove this support from these blocks here: https://github.com/WordPress/gutenberg/pull/36954

File: packages/block-library/src/comments-pagination/block.json [hunk 4]
```diff
@@ -20,7 +20,7 @@
 		"align": true,
 		"reusable": false,
 		"html": false,
-		"inserter": false,
+		"inserter": true,
```
Comment (gziolo) [inline packages/block-library/src/comments-pagination/block.json]: We can remove this line since it’s the default value.

**Comment by gziolo** [comment]
Huh, I see some failing tests completely unrelated to this PR. Do they exist on `trunk`, too?

File: packages/block-library/src/comments-pagination-next/block.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+	"$schema": "https://schemas.wp.org/trunk/block.json",
+	"apiVersion": 2,
+	"name": "core/comments-pagination-next",
+	"title": "Comments Pagination Next",
+	"category": "design",
```
Comment (gziolo) [inline packages/block-library/src/comments-pagination-next/block.json]: Let's follow other inner blocks used with the Comment Template block:
```suggestion
	"category": "theme",
```

File: packages/block-library/src/comments-pagination-next/index.php [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+	$next_comments_link = get_next_comments_link( $label, $max_page );
+
+	$wrapper_attributes = get_block_wrapper_attributes();
+
+	if ( ! isset( $next_comments_link ) ) {
+		return '';
+	}
+
+	return sprintf(
+		'<div %1s>%2s</div>',
```
Comment (gziolo) [inline packages/block-library/src/comments-pagination-next/index.php]: In the `edit` implementation we apply block wrapper attributes to the `<a>` tag. Should we align that?

File: packages/block-library/src/query-pagination-next/edit.js (hunk 3 above)
Comment (gziolo) [inline packages/block-library/src/query-pagination-next/edit.js]: Let's remove those changes from this PR, it was fixed already by @ntsekouras.

File: test/integration/fixtures/blocks/core__comments-pagination-next.html [hunk 7]
```diff
@@ -0,0 +1 @@
+<!-- wp:comments-pagination-next /-->
```
Comment (gziolo) [inline test/integration/fixtures/blocks/core__comments-pagination-next.html]: Should we include some customized attributes in the test fixtures?

File: packages/block-library/src/comments-pagination-next/index.php [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+ * Renders the `core/comments-pagination-next` block on the server.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the next comments link for the query pagination.
+ */
+function render_block_core_comments_pagination_next( $attributes, $content, $block ) {
+	$comments_per_page = isset( $block->context['queryPerPage'] ) ? $block->context['queryPerPage'] : 4;
```
Comment (gziolo) [inline packages/block-library/src/comments-pagination-next/index.php]: Where does the fallback value `4` come from?

File: packages/block-library/src/comments-pagination/edit.js [hunk 9]
```diff
@@ -37,17 +39,18 @@ export default function QueryPaginationEdit( {
 		const innerBlocks = getBlocks( clientId );
 		/**
 		 * Show the `paginationArrow` control only if a
-		 * `QueryPaginationNext/Previous` block exists.
+		 * `CommentsPaginationNext/Previous` block exists.
```
Comment (gziolo) [inline packages/block-library/src/comments-pagination/edit.js]:
```suggestion
		 * Comments Pagination Next block exists.
```

File: packages/block-library/src/comments-pagination-next/index.php (hunk 8 above)
Comment (cbravobernal) [reply packages/block-library/src/comments-pagination-next/index.php]: It was the default setting on Query Comment Loop a few days ago. As this is coming from the parent block. I think it would be good to remove it.

File: packages/block-library/src/comments-pagination/edit.js (hunk 9 above)
Comment (cbravobernal) [reply packages/block-library/src/comments-pagination/edit.js]: Let's merge first the Comments Pagination Previous. And after that we can update with:
`Comments Pagination Next and/or Previous`

**Review by gziolo — APPROVED** [review]
This PR tests well. This is what I see.

Frontend:
Screenshot 2021-12-02 at 13 05 30

Editor:
Screenshot 2021-12-02 at 13 05 40

There are some subtle differences in how the wrappers are rendered that might be related to the `<div>` vs `<a>` that I commented about.

Overall, it's good to go once all the nitpicks are addressed.

File: packages/block-library/src/comments-pagination/edit.js (hunk 9 above)
Comment (gziolo) [reply packages/block-library/src/comments-pagination/edit.js]: Whatever works best. From my perspective, we can land this PR first because the other one still needs to be reviewed 😄

**Comment by cbravobernal** [comment]
I updated the code in order to have the frontend styled. I'm not sure if we can still edit `get_query_pagination_arrow` in order to work both with comments and query loops, or it's better to create a similar function just for the comments arrow.

Attached a video with the explanation:
https://www.loom.com/share/4c9e2bdc2cba4bb08a330df59c69f45d

What do you think? 
cc @gziolo @DAreRodz @michalczaplinski @ntsekouras

**Comment by cbravobernal** [comment]
> Huh, I see some failing tests completely unrelated to this PR. Do they exist on `trunk`, too?

Seems so, is happening on other PRs as well.

**Comment by DAreRodz** [comment]
Why don't we simply use CSS to render the arrows? That way, the code won't depend on PHP functions that could change in future versions, am I right? Maybe there are some things ― RTL, accessibility, backwards compatibility, etc. ― that should be taken into consideration, and would invalidate this approach… Please, let me know. 😅 

Anyway, for example, we could use the following styles for the Comments Pagination Next block inside `styles.scss`:

```scss
> .wp-block-comments-pagination-next {
  &::after {
    margin-left: 1ch;
    display: inline-block;
  }

  &.use-arrow-chevron::after {
    content: "»";
  }
  &.use-arrow-arrow::after {
    content: "→";
    transform: scaleX(1) #{"/*rtl:scaleX(-1);*/"}; // This points the arrow right for LTR and left for RTL.
  }
}
```

…and then, in `index.php`, add the `use-arrow-arrow`, `use-arrow-chevron`, or nothing, as needed, and the correct arrow will be shown. Code logic would be simpler as well. 🤔 

```php
$arrow_attribute    = $block->context['paginationArrow'];
$wrapper_attributes = get_block_wrapper_attributes(
    array(
        'class' => "use-arrow-$arrow_attribute"
    )
);
```

**Comment by cbravobernal** [comment]
> ― RTL, accessibility, backwards compatibility, etc. ―

In this case, it seems to be an [accessibility issue](https://accessibleweb.com/question-answer/how-is-css-pseudo-content-treated-by-screen-readers/).

The main issue here is the `get_query_pagination_arrow` providing classes. Maybe an option would be to add the class with `get_block_wrapper_attributes` as you mention, in each block, and remove the class provided in that function, being more flexible.

But we have to edit that function, and I don't know if we can do it or not 😓

File: lib/compat/wordpress-5.9.1/blocks.php [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+ * @package gutenberg
+ */
+
+if ( ! function_exists( 'get_comments_pagination_arrow' ) ) {
+	/**
+	 * Helper function that returns the proper pagination arrow html for
+	 * `QueryPaginationNext` and `QueryPaginationPrevious` blocks based
+	 * on the provided `paginationArrow` from `QueryPagination` context.
+	 *
+	 * It's used in QueryPaginationNext and QueryPaginationPrevious blocks.
```
Comment (ntsekouras) [inline lib/compat/wordpress-5.9.1/blocks.php]: These are the names for `Query Pagination` blocks - if we end up using a new function.

File: lib/load.php [hunk 11]
```diff
@@ -121,6 +121,7 @@ function gutenberg_is_experiment_enabled( $name ) {
 require __DIR__ . '/compat/wordpress-5.9/class-gutenberg-rest-global-styles-controller.php';
 require __DIR__ . '/compat/wordpress-5.9/rest-active-global-styles.php';
 require __DIR__ . '/compat/wordpress-5.9/move-theme-editor-menu-item.php';
+require __DIR__ . '/compat/wordpress-5.9.1/blocks.php';
```
Comment (ntsekouras) [inline lib/load.php]: Why use `5.9.1` for this?

File: packages/block-library/src/comments-pagination-next/block.json [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+	"category": "theme",
+	"parent": [ "core/comments-pagination" ],
+	"description": "Displays the next comments page link.",
+	"textdomain": "default",
+	"attributes": {
+		"label": {
+			"type": "string"
+		}
+	},
+	"usesContext": [ "postId", "queryPerPage", "paginationArrow" ],
```
Comment (ntsekouras) [inline packages/block-library/src/comments-pagination-next/block.json]: When we are at the place of testing these blocks inside a `Query Loop` we'll need to make sure to check the context name for `queryPerPage` as we had discussed before - aliased context if needed or not.

Not a blocker of course for now.

File: packages/block-library/src/comments-pagination-next/index.php (hunk 6 above)
Comment (ntsekouras) [reply packages/block-library/src/comments-pagination-next/index.php]: Yes, we should remove the extraneous `div` here.

File: packages/block-library/src/query-pagination-next/edit.js (hunk 3 above)
Comment (ntsekouras) [reply packages/block-library/src/query-pagination-next/edit.js]: Yes, we should not forget about this 😄

**Comment by DAreRodz** [comment]
> In this case, it seems to be an [accessibility issue](https://accessibleweb.com/question-answer/how-is-css-pseudo-content-treated-by-screen-readers/).

Oh, I didn't know that. Thanks for the info, @c4rl0sbr4v0. 😊

File: packages/block-library/src/comments-pagination-next/index.php [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+	$comments_number   = (int) get_comments_number();
+	$max_page          = (int) floor( $comments_number / $comments_per_page );
+	$default_label     = __( 'Next Comments' );
+	$label             = isset( $attributes['label'] ) && ! empty( $attributes['label'] ) ? $attributes['label'] : $default_label;
+	$pagination_arrow  = get_comments_pagination_arrow( $block, true );
+	if ( $pagination_arrow ) {
+		$label .= $pagination_arrow;
+	}
+	$next_comments_link = get_next_comments_link( $label, $max_page );
```
Comment (ntsekouras) [inline packages/block-library/src/comments-pagination-next/index.php:40]: I couldn't make the pagination work as expected. I'm not sure the implementation here is correct as we have a custom `Comments query` and [get_next_comments_link](https://developer.wordpress.org/reference/functions/get_next_comments_link/) internally uses the global query `$page = get_query_var( 'cpage' );`

I guess we should take into account `get_approved_comments` that is used in the parent block to fetch the comments and we might probably need to make a new custom query ourselves, [similar to QueryPagination blocks](get_next_comments_link).

**Review by ntsekouras — COMMENTED** [review]
Thanks for your work here @c4rl0sbr4v0 !

>I'm not sure if we can still edit get_query_pagination_arrow in order to work both with comments and query loops, or it's better to create a similar function just for the comments arrow.

I think it's okay to just change the function now  since it's going to be introduced in 5.9, but we would need to backport the change to `core`. 

Having said that, I don't find it bad to actually use the existing css class as well, though. If we are planning to have this `display arrow` functionality synced, using the `.wp-block-query-pagination-` prefix, could work for both these blocks as they are actually both `query` blocks (posts/comments) and they are already wrapped inside their parent block.

File: lib/load.php (hunk 11 above)
Comment (gziolo) [reply lib/load.php]: It should use `experimental` subfolder until it gets backported to WordPress core - it’s going to be moved at some point inside `wordpress-6.0` folder if everything is ready for the major release.

Comment (cbravobernal) [reply lib/load.php]: The reason was if we had to include a new function. But I think we can use the same styles as the other arrows. Will update that!

File: packages/block-library/src/comments-pagination-next/index.php (hunk 13 above)
Comment (ntsekouras) [reply packages/block-library/src/comments-pagination-next/index.php:40]: Just noting that there were some recent changes in query pagination blocks that I'll have to check first, but the fact that I couldn't make this work as expected, still stands.

Comment (cbravobernal) [reply packages/block-library/src/comments-pagination-next/index.php:40]: I'm updating the code with a lot of changes that were requested if you prefer to wait for them. 

How is your discussion settings configured? Maybe we have a different configuration there.

Maybe we need the comment query loop working with pagination at first, although we have already the page numbers working, so I will keep taking a look at it to have it working as expected.

Regarding
> I guess we should take into account get_approved_comments that is used in the parent block to fetch the comments and we might probably need to make a new custom query ourselves, similar to QueryPagination blocks.

We started a discussion about that [right here](https://github.com/WordPress/gutenberg/issues/36642)! I think now is starting to get more relevant the requirement of having a multi-query loop for comments.

Screenshot 2021-12-03 at 16 53 19

Comment (cbravobernal) [reply packages/block-library/src/comments-pagination-next/index.php:40]: Thanks for the review! It's helping quite a lot to move the comments pagination forward.

> I couldn't make the pagination work as expected. I'm not sure the implementation here is correct as we have a custom Comments query and get_next_comments_link internally uses the global query $page = get_query_var( 'cpage' );

You are adding the block outside a post or page? If this help you, I have these Discussion settings:

Screenshot 2021-12-03 at 16 53 19

If you are not and it is not working, we should also have to take a look at the Comment Pagination, as I think we are using there also the `cpage` query var.

> I guess we should take into account get_approved_comments that is used in the parent block to fetch the comments and we might probably need to make a new custom query ourselves, similar to QueryPagination blocks.

We started a discussion [about that here](https://github.com/WordPress/gutenberg/issues/36642)! Maybe this issue will help us understand the requirement of having multi-query comments like in the Query. I will add it to the conversation.

File: packages/block-library/src/comments-pagination-next/edit.js (hunk 1 above)
Comment (cbravobernal) [reply packages/block-library/src/comments-pagination-next/edit.js]: It is not 😅 . The function that renders the arrow has that class hardcoded, so, in order to be the same both in editor and frontend and also with the same styles of the query pagination, should include that name.

File: packages/block-library/src/comments-pagination-next/index.php (hunk 2 above)
Comment (cbravobernal) [reply packages/block-library/src/comments-pagination-next/index.php]: Fixed!

File: packages/block-library/src/comments-pagination/block.json (hunk 4 above)
Comment (cbravobernal) [reply packages/block-library/src/comments-pagination/block.json]: Yep!

File: lib/compat/wordpress-5.9.1/blocks.php (hunk 10 above)
Comment (cbravobernal) [reply lib/compat/wordpress-5.9.1/blocks.php]: We will re-use the current function

File: packages/block-library/src/comments-pagination-next/block.json (hunk 12 above)
Comment (cbravobernal) [reply packages/block-library/src/comments-pagination-next/block.json]: @gziolo started working on that in #37196
We will keep an eye there!

File: packages/block-library/src/comments-pagination-next/index.php [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * @return string Returns the styles and classes defined on the block editor for the block.
+ */
+function add_next_comments_link_attributes() {
+	if ( ! function_exists( 'get_block_wrapper_attributes' ) ) {
+		return;
+	}
+	return get_block_wrapper_attributes();
+}
+add_filter( 'next_comments_link_attributes', 'add_next_comments_link_attributes' );
```
Comment (cbravobernal) [inline packages/block-library/src/comments-pagination-next/index.php]: Is this a good way to add the attributes to the `a` tag?
cc: @gziolo @ntsekouras

File: packages/block-library/src/comments-pagination-next/index.php [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+ */
+
+
+/**
+ * Applies the block attributes to the block markup without using a wrapper.
+ *
+ * @return string Returns the styles and classes defined on the block editor for the block.
+ */
+function add_next_comments_link_attributes() {
+	if ( ! function_exists( 'get_block_wrapper_attributes' ) ) {
```
Comment (gziolo) [inline packages/block-library/src/comments-pagination-next/index.php]: This function was added in WordPress 5.6 to there is no need to add the check:

https://github.com/WordPress/wordpress-develop/blob/77232140658c01121937318f0d0c1969d6827aec/src/wp-includes/class-wp-block-supports.php#L164-L169

File: packages/block-library/src/comments-pagination-next/index.php (hunk 14 above)
Comment (gziolo) [reply packages/block-library/src/comments-pagination-next/index.php]: I think I don't know the answer. Why do we need to use the filter here? Maybe we need a different implementation fo `get_next_comments_link` instead?

As far as I can tell `get_next_comments_link` works only with the singular page so it reads everything from the WordPress's page context. If we want to support custom settings for order, items per page, etc... then we need a custom implementation.

File: packages/block-library/src/comments-pagination-next/block.json (hunk 12 above)
Comment (gziolo) [reply packages/block-library/src/comments-pagination-next/block.json]: Yes, it needs to be updated. I think we should also add the namespace to the pagination arrow context value.

File: lib/blocks.php [hunk 16]
```diff
@@ -64,7 +64,9 @@ function gutenberg_reregister_core_block_types() {
 				'comment-edit-link.php'           => 'core/comment-edit-link',
 				'comment-reply-link.php'          => 'core/comment-reply-link',
 				'comment-template.php'            => 'core/comment-template',
+				'comments-pagination.php'         => 'core/comments-pagination',
```
Comment (gziolo) [inline lib/blocks.php:67]: This one is already present in `trunk`. It might fix automatically after a rebase with `trunk`.

File: packages/block-library/src/comments-pagination/block.json [hunk 17]
```diff
@@ -32,6 +32,6 @@
 			}
 		}
 	},
-	"editorStyle": "wp-block-comments-pagination-editor",
-	"style": "wp-block-comments-pagination"
+	"editorStyle": "wp-block-query-pagination-editor",
```
Comment (gziolo) [inline packages/block-library/src/comments-pagination/block.json]: Does it mean it reuses the styles from the Query Pagination block?

I'm wondering how much overlap there really is between comments and posts. For comments, I think there are now links for older/newer comments instead of next/previous page. It doesn't mean we can't change it. However, it would be great to clarify the design.

File: packages/block-library/src/comments-pagination-next/index.php (hunk 14 above)
Comment (ntsekouras) [reply packages/block-library/src/comments-pagination-next/index.php]: I haven't followed all the discussions about where the comment blocks are going, but with this approach `get_next_comments_link` and using the global query, I think this is a good way to add the attributes. That's what I did for Query pagination blocks. The thing is though that you should add the filter inside the `render` function and then you have to `remove_filter` before exiting.

File: packages/block-library/src/comments-pagination/block.json (hunk 17 above)
Comment (ntsekouras) [reply packages/block-library/src/comments-pagination/block.json]: This doesn't seem right. If we need some styles we should copy them and add the specificity of the comments block.

File: packages/block-library/src/comments-pagination-next/index.php (hunk 14 above)
Comment (cbravobernal) [reply packages/block-library/src/comments-pagination-next/index.php]: This is the only way I found to provide HTML attributes without a wrapper, directly into the `<a>` tag.

File: packages/block-library/src/comments-pagination/block.json (hunk 17 above)
Comment (cbravobernal) [reply packages/block-library/src/comments-pagination/block.json]: This reuses the styles due to [this function ](https://github.com/WordPress/gutenberg/blob/7cbba2646a522738fd95894db8240e4d3d5ac1d3/lib/compat/wordpress-5.9/blocks.php#L23), it is applying a class to the arrow with that name.

Should we create a new function to render the arrow for comments?

Could I add a new variable to set if it is a query or a comment instead?

**Comment by cbravobernal** [comment]
The PR is ready to go!

File: packages/block-library/src/comments-pagination/block.json (hunk 17 above)
Comment (ntsekouras) [reply packages/block-library/src/comments-pagination/block.json]: I'm still confused why we need a change here. And actually this results in not loading this block styles (change comments-pagination/styles.scss and check the frontend). I think we can reuse the `get_query_pagination_arrow` function which adds a css class like `wp-block-query-pagination-...`,  just copy the related styles from QueryPagination styles with any small possible needed adjustments - the styles will be wrapped under each block's base css class, and finally remove [this change](https://github.com/WordPress/gutenberg/pull/36562/files#diff-2c42b7415c23e59cddb6d85e8474729f9c2b4e3449ffc0571b3b4771e706db87R23):

`get_block_wrapper_attributes( array( 'class' => 'wp-block-query-pagination' ) )`

You made the above change for this styling issues, no? If not can you share the reason for adding it?

Comment (cbravobernal) [reply packages/block-library/src/comments-pagination/block.json]: Thanks for your patience and sorry for messing this issue so much 😓 

The reason of `get_block_wrapper_attributes( array( 'class' => 'wp-block-query-pagination' ) )` is because we are not getting the styles in editor / frontend, due to the block has the parent class `wp-block-**comment**-pagination` instead of `wp-block-**query**-pagination` on the parent.

For the arrow, the function `get_query_pagination_arrow` adds the child class `wp-block-query-pagination-$pagination_type-arrow`.

So, in order to have the correct styling, we need `.wp-block-query-pagination .wp-block-query-pagination-type-arrow`

The SCSS files on the pagination block, being the same as query pagination, won't be loaded if we only have a Comment Query Loop. That's the reason why we need them to be duplicated in both blocks.

Anyway, let's better make a small refactor to have everything clear, I'll keep you updated!

Comment (ntsekouras) [reply packages/block-library/src/comments-pagination/block.json]: No worries at all @c4rl0sbr4v0, there are a lot of subtle things here. You are doing great work here not messing anything! 😄

Comment (gziolo) [reply packages/block-library/src/comments-pagination/block.json]: The more I read about those styles, the more I think that we should build a completely independent logic for the Comments Pagination block. The styles used for Query Pagination are tied to the default class names generated for those blocks and it's only 50-ish lines of code:
https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/query-pagination/style.scss

The PHP implementation for the `get_query_pagination_arrow` is 20 lines of code:
https://github.com/WordPress/gutenberg/blob/78c7f0c7a9c3d69979d0633375325104fb03b86d/lib/compat/wordpress-5.9/blocks.php#L23-L43

It hardcodes the class name and the context entry that is tied to the Query Pagination block.
